src/app/campaign/inventory/goal-form.component.ts

```diff
@@ -37,23 +50,32 @@ export class GoalFormComponent implements OnInit, OnDestroy {
   @Output() goalChange = new EventEmitter<Flight>();
   goalForm = this.fb.group({
     id: [''],
-    totalGoal: ['', Validators.required],
-    dailyMinimum: ['']
+    totalGoal: [0, Validators.required],
+    dailyMinimum: [0],
+    uncapped: [false]
   });
   goalFormSub: Subscription;
 
+  get isCapped(): boolean {
+    return !this.goalForm.get('uncapped').value;
+  }
+
   constructor(private fb: FormBuilder) {}
 
   ngOnInit() {
     this.goalFormSub = this.goalForm.valueChanges
       .pipe(
         debounceTime(200),
         distinctUntilChanged((a, b) => {
-          return a.totalGoal === b.totalGoal && a.dailyMinimum === b.dailyMinimum;
+          return a.totalGoal === b.totalGoal && a.dailyMinimum === b.dailyMinimum && a.uncapped === b.uncapped;
         })
       )
       .subscribe(cmp => {
-        this.formStatusChanged(cmp);
+        if (cmp.uncapped) {
+          this.formStatusChanged({ ...cmp, totalGoal: 0, dailyMinimum: 0 });
```

Comment:
This is where I "reset" the goals when the user flips to "uncapped".  Let me know if it's an ngrx party foul, or if there's a better way to accomplish this.

Reply:
I think it's probably fine because it's not _supposed_ to emitEvent and we've got that extra guard on form changes...
Might suggest another way of doing it with reactive forms and rxjs by separating out the observables to each of the form component's changes. I can experiment with that as I get these forms broken up.

Reply:
+1 the more examples of correctly using reactive-forms + rxjs, the better for us all!